metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,592 @@
+)
+
+# - Referencing https://docs.datastax.com/en/cql-oss/3.x/cql/cql_using/useQuerySystem.html#Table3.ColumnsinSystem_SchemaTables-Cassandra3.0 - #
+# this keyspace contains details about the cassandra cluster's keyspaces, tables, and columns
+SYSTEM_SCHEMA_KESPACE_NAME = "system_schema"
+# these are the names of the tables we're interested in querying metadata from
+CASSANDRA_SYSTEM_SCHEMA_TABLES = {
+    "keyspaces": "keyspaces",
+    "tables": "tables",
+    "views": "views",
+    "columns": "columns",
+}
+# these column names are present on the system_schema tables
+CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES = {
+    "keyspace_name": "keyspace_name",  # present on all tables
+    "table_name": "table_name",  # present on tables table
+    "column_name": "column_name",  # present on columns table
+    "column_type": "type",  # present on columns table
+    "view_name": "view_name",  # present on views table
+    "base_table_name": "base_table_name",  # present on views table
+    "where_clause": "where_clause",  # present on views table
+}
+
+
+# -------------------------------------------------- queries -------------------------------------------------- #
+# get all keyspaces
+GET_KEYSPACES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['keyspaces']}"
+# get all tables for a keyspace
+GET_TABLES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['tables']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+# get all columns for a table
+GET_COLUMNS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['columns']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s AND {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['table_name']} = %s"
+# get all views for a keyspace
+GET_VIEWS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['views']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+
+
+# -------------------------------------------------- source config and reporter -------------------------------------------------- #
+
+# config
+class CassandraSourceConfig(PlatformInstanceConfigMixin, EnvConfigMixin):
+    contact_point: str = Field(
+        default="localhost",
+        description="The cassandra instance contact point domain (without the port).",
+    )
+    port: str = Field(default="10350", description="The cassandra instance port.")
+    username: str = Field(
+        default="",
+        description=f"The username credential. Ensure user has read access to {SYSTEM_SCHEMA_KESPACE_NAME}.",
+    )
+    password: str = Field(default="", description="The password credential.")
+    keyspace_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="regex patterns for tables to filter in ingestion.",
+    )
+
+
+# source reporter
+@dataclass
+class CassandraSourceReport(SourceReport):
```

Comment:
anything else worth adding here? other sources seem to vary in how detailed these get. what are the most common use cases for people to look at this report?